### PR TITLE
 Add "posting location" feature to message posting

### PR DESCRIPTION
**Pull request by LISH0038**
There are big changes from the previous PR (#42), so I closed #42 and created a new one (this one).

1. Fixed several suggestions from comments for #42. (Refactoring)

2. Feature flow:
a. Get the user's current location and display a neighborhood map.
b. User type in some keywords to search the location he wants on the map
c. Search results are shown by markers. 
d. User click on a marker, view the location of this marker, confirm and click "add this location"
e. Click "submit" to submit the whole message form.

3. Data store:
a. location id, name, lat and lng are passed as parameters in requests.
b. Each message only contains corresponding location_id and location_name. Location entities are stored separately. location_id is the key of Location entities for lookup.

4. Messages are displayed with corresponding location name (if contains location info).

5. **A small bug** that I cannot solve for now....
Everytime when user do search, I want the map to keep its center at user's current location. But even if I set "map.setCenter(currentPos)" at the end of the search function, the map centers till moves.
It does not affect the whole feature flow but affects the user experience...

### CODE REVIEW

**Comment by BenMQ** [comment]
wanna merge #40, and then edit this to be targeting master branch? that will make the diff easier to read

**Review by BenMQ — COMMENTED** [review]
Mostly look good! i'm slightly confused, how is location_id generated? I see that there's a hidden field but not understanding what's populating it / how will it be used.

File: src/main/webapp/js/map-loader.js [hunk 1]
```diff
@@ -1,29 +1,180 @@
...
+  function buildInfoWindowWithSubmit(place){
+    const position = place.geometry.location;
+
+    const para = document.createElement("P");
+    const text = document.createTextNode(place.name);
+    para.appendChild(text);
+    const button = document.createElement('button');
+    button.appendChild(document.createTextNode('Add this location'));
+    button.onclick = () => {
+      addLocationInfoToForm(position.lat(), position.lng(), place.name, place.place_id);
```
Comment (LISH0038) [inline src/main/webapp/js/map-loader.js:91]: @BenMQ  Here's the logic of populating values for hidden inputs. When the button "Add this location" is hit, this function "addLocationInfoToForm" creates hidden inputs, which are meant to be submitted to servlet along with the form later. Here place.place_id is passed in as location_id, it is an id maintained by google map api, which uniquely identifies each location on google map. The search result "place" contains this information so I just directly take it.

Comment (BenMQ) [reply src/main/webapp/js/map-loader.js:91]: good to know, thanks for explaining!

**Review by BenMQ — APPROVED** [review]

**Pull request merged**